app/assets/javascripts/request_refund.js

```diff
@@ -0,0 +1,34 @@
+$(document).ready(function () {
+  $('.refund-button').click(function () {
+    var uniq_id = $(this).data('id');
+    $('#uniq_id').val(uniq_id);
+  });
+  $('#reason-label').hide();
+  $('input[type="radio"]').click(function () {
+    if ($('#reason-other').is(':checked')) {
+      $('#reason-label').show();
+    } else {
+      $('#reason-label').hide();
+    }
```

Comment:
Great refactoring. However, value of `option_state` is not updated based on the click event on line `7`